src/runtime/types.h

```diff
@@ -371,14 +375,31 @@ class BoxedBool : public BoxedInt {
 
 class BoxedString : public Box {
 public:
-    // const std::basic_string<char, std::char_traits<char>, StlCompatAllocator<char> > s;
-    std::string s;
+    llvm::StringRef s;
+
+    char* storage() { return (char*)this + cls->tp_basicsize; }
+
+    void* operator new(size_t size, size_t ssize) __attribute__((visibility("default"))) {
+        Box* rtn = static_cast<Box*>(gc_alloc(str_cls->tp_basicsize + ssize + 1, gc::GCKind::PYTHON));
+        rtn->cls = str_cls;
+        return rtn;
+    }
 
+    void* operator new(size_t size, BoxedClass* cls, size_t ssize) __attribute__((visibility("default"))) {
+        // XXX this is likely broken...
```

Comment:
ah, because I hadn't verified that `tp_itemsize == sizeof(char)` in cpython.  just did that, and removing comment.